appshell/appshell_extensions_platform.cpp

```diff
@@ -0,0 +1,29 @@
+#include "appshell/appshell_extensions_platform.h"
+#include <unicode/ucsdet.h>
+
+void GetCharsetMatch(const char* bufferData, size_t bufferLength, std::string &detectedCharSet) {
+	const UCharsetMatch* charsetMatch_;
+	UErrorCode icuError = U_ZERO_ERROR;
+
+	UCharsetDetector* charsetDetector_ = ucsdet_open(&icuError);
+	if (U_FAILURE(icuError))
+		throw "Failed to open detector";
+
+	// send text
+	ucsdet_setText(charsetDetector_, bufferData, bufferLength, &icuError);
+	if (U_FAILURE(icuError))
+		throw "Failed to set text";
```

Comment:
I'm not sure how this would work out. From this function, it appears detectedCharSet seems to be the output parameter. Should we initialize the detectedCharSet to empty string in this function at the very beginning?